flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -121,4 +135,36 @@ protected Catalog createCatalog(String name, Map<String, String> properties, Con
   public static Configuration clusterHadoopConf() {
     return HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
   }
+
+  private void loadHiveConf(Configuration configuration, Map<String, String> properties) {
+    String hiveConfPath = properties.get(HIVE_SITE_PATH);
+    Path path = new Path(hiveConfPath);
+    if (hiveConfPath.startsWith("hdfs")) {
+      download(configuration, path);
+    } else {
+      File file = new File(hiveConfPath);
+      if (!file.exists()) {
+        LOGGER.error("{} doesn't exist. if in application mode ," +
+                " please provide a hdfs path for hive-site.xml", hiveConfPath);
+      } else {
+        configuration.addResource(path);
+      }
+    }
+  }
+
+  private void download(Configuration configuration, Path hdfsHiveSitePath) {
+    try {
+      File tmpFile = File.createTempFile("hive-site.xml-", "");
```

Comment:
Is the temp file `hive-site.xml-` supposed to have that trailing dash? Is this possibly a typo?

Reply:
My initial idea was to add a prefix to distinguish it from the actual hive-site.xml. We can add a .tmp suffix to make users understand that it is only a temporary file

Reply:
That sounds much better to me. `.tmp` definitely better conveys the intent to me. Thanks for updating that!